### PR TITLE
update face-attributes to latest conventions

### PR DESCRIPTION
**Pull request by lebensterben**
Changed `nil` values in `:foreground` and `:background` attributes to `'unspecified`.
Changed `:bold t` and `:bold nil` to `:weight 'bold` and `:weight 'normal` respectively.
Changed `:italic t` and `:italic nil` to `:slant 'italic` and `:slant 'normal` respectively.

### CODE REVIEW

File: spacemacs-common.el [hunk 1]
```diff
@@ -611,15 +611,15 @@ to 'auto, tags may not be properly aligned. "
...
      `(font-latex-sectioning-1-face ((,class (:inherit bold :foreground ,head4 :height ,(if spacemacs-theme-org-height 1.3 1.0) :background ,(when spacemacs-theme-org-highlight head4-bg)))))
      `(font-latex-sectioning-2-face ((,class (:inherit bold :foreground ,head1 :height ,(if spacemacs-theme-org-height 1.3 1.0) :background ,(when spacemacs-theme-org-highlight head1-bg)))))
      `(font-latex-sectioning-3-face ((,class (:inherit bold :foreground ,head2 :height ,(if spacemacs-theme-org-height 1.2 1.0) :background ,(when spacemacs-theme-org-highlight head2-bg)))))
-     `(font-latex-sectioning-4-face ((,class (:bold nil :foreground ,head3 :height ,(if spacemacs-theme-org-height 1.1 1.0) :background ,(when spacemacs-theme-org-highlight head3-bg)))))
-     `(font-latex-sectioning-5-face ((,class (:bold nil :foreground ,head4 :background ,(when spacemacs-theme-org-highlight head4-bg)))))
+     `(font-latex-sectioning-4-face ((,class (:weight normal :foreground ,head3 :height ,(if spacemacs-theme-org-height 1.1 1.0) :background ,(when spacemacs-theme-org-highlight head3-bg)))))
```
Comment (eeshugerman) [inline spacemacs-common.el:621]:
```suggestion
     `(font-latex-sectioning-4-face ((,class (:weight normal :foreground ,head3 :height ,(if spacemacs-theme-org-height 1.1 1.0) :background ,(if spacemacs-theme-org-highlight head3-bg 'unspecified)))))
```

This goes for the other `when` expressions too.

**Comment by lebensterben** [comment]
Updated

**Review by real-or-random — APPROVED** [review]
diff looks good to me but I haven't tested it

**Review by eeshugerman — APPROVED** [review]
I've been running this for a few hours now. Warning are gone and I don't see any breakage.

**Comment by iwahbe** [comment]
Is there any ETA for merging this?

**Comment by nashamri** [comment]
Thanks all for this, and sorry for being late.

**Pull request merged**